### PR TITLE
Refactor KeybasePacket to move it out of libkb

### PR DESCRIPTION
**Pull request by akalin-keybase**
It had a dependence on SKB, which doesn't need to be moved out,
so break that dependency, using the fact that when we're decoding
a keybase packet, we always know what kind of packet we expect.

This also saves on having to reencode some stuff, which is nice.

### CODE REVIEW

**Comment by akalin-keybase** [comment]
This is (1/3) in a series, next one is https://github.com/keybase/client/pull/13514

File: go/libkb/skb_test.go [hunk 1]
```diff
@@ -285,11 +234,7 @@ func testErrUnlock(t *testing.T, skb *SKB, ui *TestCancelSecretUI) error {
 		UseCancelCache: true,
 	}
 	key, err := skb.PromptAndUnlock(NewMetaContextTODO(skb.G()), parg, NewSecretStore(skb.G(), "testusername"), nil)
-	if err == nil {
-		t.Fatal("PromptAndUnlock returned nil error")
-	}
-	if key != nil {
-		t.Errorf("PromptAndUnlock returned a key (%v)", key)
-	}
+	require.NotNil(t, err)
```
Comment (maxtaco) [inline go/libkb/skb_test.go]: `NotError`

Comment (akalin-keybase) [reply go/libkb/skb_test.go]: I think you mean `Error`...fixed!

**Comment by akalin-keybase** [comment]
Thanks, Max! Merging after CI and after I do a review pass.

**Comment by akalin-keybase** [comment]
Actually I'll probably wait until the release completes.

**Pull request merged**